checkstyle.xml

```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE module PUBLIC
+        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
+        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
+<!--
+DSpace CodeStyle Requirements
+
+  1. 4-space indents for Java, and 2-space indents for XML. NO TABS ALLOWED.
+     (Only exception is throws clause, which should be indented 8 spaces if on a new line)
+  2. K&R style braces required. Braces required on all blocks.
+  3. Do not use wildcard imports (e.g. import java.util.*). Duplicated or unused imports also not allowed.
+  4. Javadocs required for all public classes and methods. Keep it short and to the point
+  5. Maximum line length is 120 characters
+  6. No trailing spaces allowed (except in comments)
+  7. Tokens should be surrounded by whitespace (see http://checkstyle.sourceforge.net/config_whitespace.html#WhitespaceAround)
+  8. Each source file must include our license header (validated separately by license-maven-plugin, see pom.xml)
+
+For more information on CheckStyle configurations below, see: http://checkstyle.sourceforge.net/checks.html
+-->
+<module name="Checker">
+    <!-- Configure checker to use UTF-8 encoding -->
+    <property name="charset" value="UTF-8"/>
+    <!-- Configure checker to run on files with these extensions -->
+    <property name="fileExtensions" value="java, properties, cfg, xml"/>
+
+    <!-- No tab characters ('\t') allowed in the source code -->
+    <module name="FileTabCharacter">
+        <property name="eachLine" value="true"/>
+        <property name="fileExtensions" value="java, properties, cfg, css, js, xml"/>
+    </module>
+
+    <!-- No Trailing Whitespace, except on lines that only have an asterisk (e.g. Javadoc comments) -->
+    <module name="RegexpSingleline">
+        <property name="format" value="(?&lt;!\*)\s+$|\*\s\s+$"/>
+        <property name="message" value="Line has trailing whitespace"/>
+        <property name="fileExtensions" value="java, properties, cfg, css, js, xml"/>
+    </module>
+
+    <!-- Allow individual lines of code to be excluded from these rules, if they are annotated
+         with @SuppressWarnings. See also SuppressWarningsHolder below -->
+    <module name="SuppressWarningsFilter" />
+
+    <!-- Check individual Java source files for specific rules -->
+    <module name="TreeWalker">
+        <!-- Maximum line length is 120 characters -->
+        <module name="LineLength">
+            <property name="max" value="120"/>
```

Comment:
I want only note that on the current javadoc for a REST class we have a violation that cannot easily solved...
https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/utils/MultipartFileSender.java#L31

due to the long URL...

Reply:
Good point, @abollini.  It looks like we could simply tweak this setting similar to how Google does, and add in an `ignorePattern` exception for long URLs:
https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml#L44

Reply:
This has now been resolved with exceptions for long URLs, packages or imports.